docs/arch-overview.md

```diff
@@ -0,0 +1,82 @@
+# K8ssandraCluster Components
+K8ssandraCluster is the primary custom resource with which you will with interact when using K8ssandra Operator. This document provides a high level overview of the K8ssandraCluster custom resource. 
+
+The following diagram highlights the key objects that comprise a K8ssandraCluster. The K8ssandraCluster spec is shown with highlighted parts pointing to corresponding objects.
+
+![](img/arch-overview.png)
+
+# Kubernetes Clusters
+The various components that make up a K8ssandraCluster can be spread across multiple Kubernetes clusters. One of those clusters must be designated as the control plane cluster. The other clusters are data plane clusters. In the diagram we have three Kubernetes clusters - `control-plane`, `east`, and `west`. The latter two make up the data plane.
+
+The K8ssandraCluster object must be created in the control plane cluster.
+
+K8ssandra Operator is deployed in each cluster. The operator consists of several controllers. Some controllers, notably the K8ssandraCluster controller, only run in the control plane cluster. This is the main reason that the K8ssandraCluster object needs be created in the control plane cluster. 
+
+# CassandraDatacenter
+Here we discuss the `cassandra.datacenters` property which is an array. K8ssandra Operator creates a CassandraDatacenter for each element. 
+
+Notice the `name: dc1` that is highlighted in green. This tells K8ssandra Operator to create a CassandraDatacenter named `dc1`. The green arrow in the diagram points to the CassandraDatacenter object. The `size` property declares that there should be 3 Cassandra nodes.
+
+<!-- TODO: Add link to configuration doc when its available -->
+
+Cass Operator manages the CassandraDatacenter. Note the arrow from the CassandraDatacenter to the StatefulSet `test-dc1-default-sts`. Cass Operator creates the StatefulSet. The StatefulSet controller, which is part of Kubernetes itself, creates the actual pods that run Cassandra.
+
+
+## k8sContext
+K8ssandra Operator creates `dc1` in the `east` cluster. The `k8sContext` property determines in which Kubernetes cluster the operator creates the CassandraDatacenter. If not specified the operator creates the CassandraDatacenter in the control plane cluster.
```

Comment:
So this comment applies all over I think, but as I'm reading through it, i think it might be nice to have textual examples of the spec's you're talking about referencing in the diagram embedded with each section.  It saves some scrolling back and forth, especially as you get to content further down the screen.